app/gui/src/controller/searcher.rs

```diff
@@ -939,91 +642,45 @@ impl Searcher {
-    #[profile(Debug)]
-    fn make_action_list(
-        &self,
-        completion_response: &language_server::response::Completion,
-    ) -> action::List {
-        let creating_new_node = matches!(self.mode.deref(), Mode::NewNode { .. });
-        let should_add_additional_entries = creating_new_node && self.this_arg.is_none();
-        let mut actions = action::ListWithSearchResultBuilder::new();
-        let (libraries_icon, default_icon) =
-            action::hardcoded::ICONS.with(|i| (i.libraries.clone_ref(), i.default.clone_ref()));
-        if should_add_additional_entries && self.ide.manage_projects().is_ok() {
-            let mut root_cat = actions.add_root_category("Projects", default_icon.clone_ref());
-            let category = root_cat.add_category("Projects", default_icon.clone_ref());
-            let create_project = action::ProjectManagement::CreateNewProject;
-            category.add_action(Action::ProjectManagement(create_project));
-        }
-        let mut libraries_root_cat =
-            actions.add_root_category("Libraries", libraries_icon.clone_ref());
-        if should_add_additional_entries {
-            let examples_cat =
-                libraries_root_cat.add_category("Examples", default_icon.clone_ref());
-            examples_cat.extend(self.database.iterate_examples().map(Action::Example));
-        }
-        let libraries_cat =
-            libraries_root_cat.add_category("Libraries", libraries_icon.clone_ref());
-        let entries = completion_response.results.iter().filter_map(|id| {
-            self.database
-                .lookup(*id)
-                .map(|entry| Action::Suggestion(action::Suggestion::FromDatabase(entry)))
-                .handle_err(|e| {
-                    error!(
-                        "Response provided a suggestion ID that cannot be \
-                    resolved: {e}."
-                    )
-                })
+            };
+            this.data.borrow_mut().components = new_list;
+            this.notifier.publish(Notification::NewComponentList).await;
         });
-        libraries_cat.extend(entries);
-
-        actions.build()
     }
 
     #[profile(Debug)]
     fn make_component_list<'a>(
         &self,
         entry_ids: impl IntoIterator<Item = suggestion_database::entry::Id>,
         this_type: &Option<String>,
-    ) -> component::List {
-        let favorites = self.graph.component_groups();
-        let module_name = self.module_qualified_name();
-        let mut builder = component_list_builder_with_favorites(
-            &self.database,
-            module_name.as_ref(),
-            &*favorites,
-            self.ide.are_component_browser_private_entries_visible(),
-        );
-        add_virtual_entries_to_builder(&mut builder, this_type);
-        builder.extend_list_and_allow_favorites_with_ids(&self.database, entry_ids);
-        builder.build()
+    ) -> Rc<component::List> {
```

Comment:
I see there are some `Rc<component::List>`. Could we simplify the usage of this type by using the pattern where `component::List` wraps an internal, never-cloned type in an `Rc`? It would allow eliminating a lot of `Rc`s inside `List`.

Reply:
I'm not convinced. I know we do this a lot in other places, but this actually adds a layer of understanding what's inside list (we have two structures to learn instead of one), I also would need to implement `make_mut` equivalent and explain how it works. 

It's actually more complicated than a situation where programmer see `Rc` and instantly see what's going on.